b2sdk/transfer/emerge/unbound_write_intent.py

```diff
@@ -0,0 +1,230 @@
+######################################################################
+#
+# File: b2sdk/transfer/emerge/unbound_write_intent.py
+#
+# Copyright 2022 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+import hashlib
+import io
+import queue
+from typing import (
+    Callable,
+    Iterator,
+    Optional,
+    Union,
+)
+
+from b2sdk.exception import B2SimpleError
+from b2sdk.transfer.emerge.write_intent import WriteIntent
+from b2sdk.transfer.outbound.upload_source import AbstractUploadSource
+
+
+class UnboundStreamBufferTimeout(B2SimpleError):
+    """
+    Raised when there is no space for a new buffer for a certain amount of time.
+    """
+    pass
+
+
+class IOWrapper(io.BytesIO):
+    """
+    Wrapper for BytesIO that knows when it has been read in full.
+
+    When created, it tries to put itself into a waiting queue,
```

Comment:
It doesn't seem to do any of that

Reply:
Good catch! It's a comment from halfway through.